### PR TITLE
Change position to toast android

### PR DESCRIPTION
**Pull request by yaxter**
### Description of Changes
* ...
* Now it is possible to change the position in which the toast android is displayed
* ...

### CODE REVIEW

File: kivymd/toast/androidtoast/androidtoast.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    toast.show()
+
+
+# text: Text to be displayed in the toast
+# short_duration: duration of the toast, if True the toast will last 2.3s but if it is False the toast will last 3.9s
+# gravity: refers to the toast position, if it is 80 the toast will be shown below, if it is 40 the toast will be displayed above
+# y: refers to the vertical position of the toast
+# x: refers to the horizontal position of the toast
+
+# important: if only the text value is specified and the value of the gravity, y, x parameters is not specified, their values ​​will be 0 which means that the toast will be shown in the center
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Place docstrings in the method itself.

File: kivymd/toast/androidtoast/androidtoast.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        
+    	
+class Iniciar(MDApp):
+    def build(self):
+        pantalla = ScreenManager()
+        screen = Screen1(name="uno")
+        pantalla.add_widget(screen)
+        return pantalla
+
+Iniciar().run()
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Use markup format:

.. code-block:: python

    from kivymd.toast import toast
    import kivy, kivymd
    from kivymd.app import MDApp
    from kivy.lang import Builder
    from kivy.uix.screenmanager import Screen, ScreenManager
    
    Builder.load_string('''
    <Screen1>:
        MDFlatButton:
            text:"My Toast"
            pos_hint:{"center_x": .5, "center_y": .05}
            on_press:root.show_toast()
    ''')
    
    
    class Screen1(Screen):
        def show_toast(self):
            toast("hola cómo estan", True, 80, 200, 0)
    
    
    class Iniciar(MDApp):
        def build(self):
            pantalla = ScreenManager()
            screen = Screen1(name="uno")
            pantalla.add_widget(screen)
            return pantalla
    
    
    Iniciar().run()

File: kivymd/toast/androidtoast/androidtoast.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+"""
+.. rubric:: Native implementation of toast for Android devices.
+.. code-block:: python
+    # Will be automatically used native implementation of the toast
+    # if your application is running on an Android device.
+    # Otherwise, will be used toast implementation
+    # from the kivymd/toast/kivytoast package.
+from kivymd.toast import toast
+import  kivy,kivymd
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Remove unused imports.

File: kivymd/toast/androidtoast/androidtoast.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+Builder.load_string('''
+<Screen1>:
+    MDFlatButton:
+        text:"My Toast"
+        pos_hint:{"center_x": .5, "center_y": .05}
+        on_press:root.show_toast()
+''')
+
+class Screen1(Screen):
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Use `MDScreen` not `Screen`

File: kivymd/toast/androidtoast/androidtoast.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    def show_toast(self):
+        toast("hola cómo estan",True,80,200,0)
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Fix PEP8

File: kivymd/toast/androidtoast/androidtoast.py (hunk 4 above)
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Bad name for class

**Review by HeaTTheatR — CHANGES_REQUESTED** [review]

**Comment by yaxter** [comment]
Sorry friend is that I hardly use GitHub, but I already fixed what you told me

File: kivymd/toast/androidtoast/androidtoast.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
+"""
+
+#Will be automatically used native implementation of the toast
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Make one space

File: kivymd/toast/androidtoast/androidtoast.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+# if your application is running on an Android device.
+# Otherwise, will be used toast implementation
+# from the kivymd/toast/kivytoast package.
+
+from kivymd.toast import toast
+from kivymd.app import MDApp
+from kivy.lang import Builder
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: There should be one indent

File: kivymd/toast/androidtoast/androidtoast.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+from kivy.uix.screenmanager import ScreenManager
+from kivymd.uix.screen import MDScreen
+
+
+
+
+Builder.load_string('''
+<Example>:
+    MDFlatButton:
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: There should be one indent

File: kivymd/toast/androidtoast/androidtoast.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+from kivymd.app import MDApp
+from kivy.lang import Builder
+from kivy.uix.screenmanager import ScreenManager
+from kivymd.uix.screen import MDScreen
+
+
+
+
+Builder.load_string('''
+<Example>:
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Remove punctuation mark colon

File: kivymd/toast/androidtoast/androidtoast.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    MDFlatButton:
+        text:"My Toast"
+        pos_hint:{"center_x": .5, "center_y": .05}
+        on_press:root.show_toast()
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: PEP8

File: kivymd/toast/androidtoast/androidtoast.py [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+''')
+
+class Example(MDScreen):
+    def show_toast(self):
+        toast("hola cómo estan",True,80,200,0)
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: PEP8

File: kivymd/toast/androidtoast/androidtoast.py [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+@run_thread
+def toast(text,short_duration=True,gravity=0,y=0,x=0):
+    
+    #text: Text to be displayed in the toast
+    #short_duration: duration of the toast, if True the toast will last 2.3s but if it is False the toast will last 3.9s
+    #gravity: refers to the toast position, if it is 80 the toast will be shown below, if it is 40 the toast will be displayed above 
+    #y: refers to the vertical position of the toast
+    #x: refers to the horizontal position of the toast	
+      
+    #important: if only the text value is specified and the value of the gravity, y, x parameters is not specified, their values ​​will be 0 which means that the toast will be shown in the center
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python
    def toast(text, short_duration=True, gravity=0, y=0, x=0):
        """
        
        :param short_duration:
        :param gravity:
        :param y:
        :param x:
        :return:
        """
```

File: kivymd/toast/androidtoast/androidtoast.py [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    #y: refers to the vertical position of the toast
+    #x: refers to the horizontal position of the toast	
+      
+    #important: if only the text value is specified and the value of the gravity, y, x parameters is not specified, their values ​​will be 0 which means that the toast will be shown in the center  
+    
+    duration = __Toast.LENGTH_SHORT if short_duration else __Toast.LENGTH_LONG  
+      
+    toast = __Toast.makeText(__activity, __String(text), duration)
+        	
+    toast.setGravity(gravity,x,y)
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: PEP8

**Review by HeaTTheatR — CHANGES_REQUESTED** [review]

File: kivymd/toast/androidtoast/androidtoast.py (hunk 13 above)
Comment (yaxter) [reply kivymd/toast/androidtoast/androidtoast.py]: I don't know what you mean by pep8

Comment (HeaTTheatR) [reply kivymd/toast/androidtoast/androidtoast.py]: https://github.com/kivymd/KivyMD#setup-environment

Comment (yaxter) [reply kivymd/toast/androidtoast/androidtoast.py]: F

Comment (yaxter) [reply kivymd/toast/androidtoast/androidtoast.py]: 😔

File: kivymd/toast/androidtoast/androidtoast.py [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    
+    from kivymd.toast import toast
+    from kivymd.app import MDApp
+    from kivy.lang import Builder
+    from kivy.uix.screenmanager import ScreenManager
+    from kivymd.uix.screen import MDScreen
+    
+    
+    
+
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py:16]: Why are you putting four indents here? There must be one.

File: kivymd/toast/androidtoast/androidtoast.py [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+    
+    
+    
+    
+    Builder.load_string('''
+    <Example>
+        MDFlatButton:
+            text:"My Toast"
+            pos_hint:{"center_x": .5, "center_y": .05}
+            on_press:root.show_toast()
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```
    <Example>

        MDFlatButton:
            text: "My Toast"
            pos_hint: {"center_x": .5, "center_y": .05}
            on_press: root.show_toast()
```

File: kivymd/toast/androidtoast/androidtoast.py [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+    ''')
+    
+    class Example(MDScreen):
+        def show_toast(self):
+            toast("hello world",True,80,200,0)
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python
    class Example(MDScreen):
        def show_toast(self):
            toast("hello world", True, 80, 200, 0)
```

File: kivymd/toast/androidtoast/androidtoast.py [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+            return pantalla
+    
+    Iniciar().run()
+"""
+__all__ = ("toast",)
+
+from jnius import autoclass    
+from android.runnable import run_on_ui_thread as run_thread
+
+
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]: Why are you putting two indents here? There must be one.

File: kivymd/toast/androidtoast/androidtoast.py [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+    
+    Iniciar().run()
+"""
+__all__ = ("toast",)
+
+from jnius import autoclass    
+from android.runnable import run_on_ui_thread as run_thread
+
+        
+__activity =autoclass( 'org.kivy.android.PythonActivity').mActivity
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python
__activity = autoclass('org.kivy.android.PythonActivity').mActivity
```

File: kivymd/toast/androidtoast/androidtoast.py [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+def toast(text,short_duration=True,gravity=0,y=0,x=0):
+    """        
+    : param text Text to be displayed in the toast
+    :: param short_duration  duration of the toast, if True the toast will last 2.3s but if it is False the toast will last 3.9s
+    :: param gravity refers to the toast position, if it is 80 the toast will be shown below, if it is 40 the toast will be displayed above 
+    :: param y refers to the vertical position of the toast
+    :: param x  refers to the horizontal position of the toast	
+      
+    :: important  if only the text value is specified and the value of the gravity, y, x parameters is not specified, their values ​​will be 0 which means that the toast will be shown in the center
+"""
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python
    def toast(text, short_duration=True, gravity=0, y=0, x=0):
        """
        :param text: text to be displayed in the toast;
        :param short_duration:  duration of the toast, if `True` the toast
               will last 2.3s but if it is `False` the toast will last 3.9s;
        :param gravity: refers to the toast position, if it is 80the toast will
               be shown below, if it is 40 the toast will be displayed above;
        :param y: refers to the vertical position of the toast;
        :param x: refers to the horizontal position of the toast;

        Important: if only the text value is specified and the value of
        the `gravity`, `y`, `x` parameters is not specified, their values ​​will
        be 0 which means that the toast will be shown in the center.
        """
```

File: kivymd/toast/androidtoast/androidtoast.py [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+
+
+  
+    duration = __Toast.LENGTH_SHORT if short_duration else __Toast.LENGTH_LONG  
+      
+    toast = __Toast.makeText(__activity, __String(text), duration)
+        	
+    toast.setGravity(gravity,x,y)
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python 
    duration = __Toast.LENGTH_SHORT if short_duration else __Toast.LENGTH_LONG  
    toast = __Toast.makeText(__activity, __String(text), duration)
    toast.setGravity(gravity, x, y)
```

**Review by HeaTTheatR — CHANGES_REQUESTED** [review]

File: kivymd/toast/androidtoast/androidtoast.py (hunk 14 above)
Comment (yaxter) [reply kivymd/toast/androidtoast/androidtoast.py:16]: I made 1 tabulation, it is also a comment does not affect the operation, the code works and that is the important thing

File: kivymd/toast/androidtoast/androidtoast.py (hunk 20 above)
Comment (yaxter) [reply kivymd/toast/androidtoast/androidtoast.py]: And if you fix it

File: kivymd/toast/androidtoast/androidtoast.py (hunk 14 above)
Comment (HeaTTheatR) [reply kivymd/toast/androidtoast/androidtoast.py:16]: No. The code must be written according to PEP8 standards - https://www.python.org/dev/peps/pep-0008/

**Comment by yaxter** [comment]
forgive my ignorance, I didn't know it was pep8

**Comment by yaxter** [comment]
I think everything is fixed, I also formatted it with black

File: kivymd/toast/androidtoast/androidtoast.py [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+     
+     def build(self):
+         pantalla = ScreenManager()
+         screen = Example(name="one")
+         pantalla.add_widget(screen)
+         return pantalla
+    
+ Iniciar().run()
+"""
+__all__ = ("toast",)
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python
"""
[...]

 class Example(MDScreen):
     def show_toast(self):
         toast("hello world", True, 80, 200, 0) 

 class Iniciar(MDApp):
     def build(self):
         pantalla = ScreenManager()
         screen = Example()
         pantalla.add_widget(screen)
         return pantalla
    
 Iniciar().run()
"""

__all__ = ("toast",)
```

File: kivymd/toast/androidtoast/androidtoast.py [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+from android.runnable import run_on_ui_thread as run_thread
+
+
+__activity = autoclass("org.kivy.android.PythonActivity").mActivity
+__Toast = autoclass("android.widget.Toast")
+__String = autoclass("java.lang.String")
+
+
+@run_thread
+def toast(text, short_duration=True, gravity=0, y=0, x=0):
```
Comment (HeaTTheatR) [inline kivymd/toast/androidtoast/androidtoast.py]:
```python
from jnius import autoclass
from android.runnable import run_on_ui_thread as run_thread

__activity = autoclass("org.kivy.android.PythonActivity").mActivity
__Toast = autoclass("android.widget.Toast")
__String = autoclass("java.lang.String")


@run_thread
def toast(text, short_duration=True, gravity=0, y=0, x=0):
```

**Review by HeaTTheatR — CHANGES_REQUESTED** [review]

**Comment by yaxter** [comment]
:/

**Pull request merged**